projects/swimlane/ngx-charts/src/lib/common/axes/x-axis-ticks.component.ts

```diff
@@ -79,10 +85,16 @@ export class XAxisTicksComponent implements OnChanges, AfterViewInit {
   }
 
   updateDims(): void {
+    if (!isPlatformBrowser(this.platformId)) {
+      // for SSR, use approximate value instead of measured
+      this.dimensionsChanged.emit({ height: this.approxHeight });
+      return;
+    }
+
     const height = parseInt(this.ticksElement.nativeElement.getBoundingClientRect().height, 10);
     if (height !== this.height) {
       this.height = height;
-      this.dimensionsChanged.emit({ height });
+      this.dimensionsChanged.emit({ height: this.height });
```

Comment:
Is this needed?

Reply:
Thought it's cleaner to emit the actual property value.